File: frontend/src/app/dto/gameRoom.ts
import {GameRoomStatusTypes} from './enums/gameRoomStatusTypes';

export class GameRoom {
  id: number;
  code: string;
  type: string;
  name: string;
  round: number;
  status: GameRoomStatusTypes;
  question: number;
  playerId: number;
  timeRemaining: number;

  constructor(args: GameRoom = <GameRoom>{}) {
    this.id = args.id;
    this.code = args.code;
    this.type = args.type;
    this.name = args.name;
    this.round = args.round;
    this.status = args.status;
    this.question = args.question;
    this.playerId = args.playerId;
    this.timeRemaining = args.timeRemaining;
  }
}

File: src/main/java/com/corelogic/schemaconverter/entity/GameRoom.java
package com.corelogic.schemaconverter.entity;

import com.corelogic.schemaconverter.entity.enums.GameRoomStatus;
import com.corelogic.schemaconverter.entity.enums.GameRoomType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class GameRoom {

    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "GameRoomSeq")
    @SequenceGenerator(name = "GameRoomSeq", sequenceName = "game_room_id_seq", allocationSize = 1)
    @Id
    private Long id;

    private String code;
    @Enumerated(EnumType.STRING)
    private GameRoomType type;
    private String name;
    private Integer round;
    @Enumerated(EnumType.STRING)
    private GameRoomStatus status;

    private long question;
    private long playerId;
}

